pennylane/compiler/compiler.py

```diff
@@ -18,6 +18,12 @@
 from importlib import reload, metadata
 from collections import defaultdict
 import dataclasses
+import os
+import re
+
+from semantic_version import Version
+
+PL_CATALYST_MIN_VERSION = Version(os.environ.get("PL_CATALYST_MIN_VERSION", "0.4.0"))
```

Comment:
An environment variable determining a minimum package version is also a bit strange. Are there alternatives we can think of?

Maybe temporarily patching this value, or the version value of the installed Catalyst package, in the test configuration, until after the release? The post-release update would only touch the tests then, without needing to touch the package source.

Reply:
@dime10 I only used this env variable to avoid adding any monkey patches to PL (tests) as this sort of patching should be prohibited in every codebase if possible. But, if you think monkey patching the version of Catalyst is something "okay" until the v0.4.0 release, I'd be happy to adjust the code.

Reply:
I don't agree monkey patching should necessarily be prohibited in every code base, in fact it is a common pattern for testing (https://docs.pytest.org/en/latest/how-to/monkeypatch.html), which is the use case we are discussing.
So personally I would favour it over an environment variable, but I'm also happy to let core decide here.

Reply:
Thanks @dime10! I'll update the code. The core team is okay with this temporary monkey patch :+1: